### PR TITLE
[#105] [Part 2] Add OpenAPI to support creating API documentation with OpenAPI

### PR DESCRIPTION
**Pull request by suho**
#105

## What happened 👀

- Added an endpoint (`/api/docs/openapi`) to support generating OpenAPI docs by using https://github.com/stoplightio/elements

## Insight 📝

This is `Part 2` for the #105 issue. I configured https://github.com/stoplightio/elements to visualize the OpenAPI into the gin templates. Here is the checklist of #105 when this pull request is merged:

- [x] Add OpenAPI addon as an optional addon in the template.
- [x] Add OpenAPI tooling (Spectral, Elements, Lint).
- [x] Create an endpoint (e.g., GET {{base_url}}/api/docs/openapi) to access the API documentation through the browser.
- [ ] Add Mock Server with Prims (related to this [API Mock Server with OpenAPI](https://www.notion.so/API-Mock-Server-with-OpenAPI-83fda1653eb1409fab9b08c1fcf466fd?pvs=21)).
- [ ] Add CI/CD workflow to run linting and deploy mock server to fly.io:
  - [x] Linting
  -	[ ] Deploy Mock Server

## Proof Of Work 📹

Screenshot 2023-08-14 at 15 15 17

### CODE REVIEW

**Review by olivierobert — APPROVED** [review]

File: {{cookiecutter.app_name}}/lib/api/docs/routers/router.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package routers
+
+import (
+	"github.com/nimblehq/{{cookiecutter.app_name}}/lib/api/docs/controllers"
+
+	"github.com/gin-gonic/gin"
+)
+
+func ComebineRoutes(engine *gin.Engine) {
```
Comment (Nihisil) [inline {{cookiecutter.app_name}}/lib/api/docs/routers/router.go]: is it typo?
```suggestion
func CombineRoutes(engine *gin.Engine) {
```

and combine routes name itself sounds strange for me 🤔

File: {{cookiecutter.app_name}}/lib/api/docs/views/openapi/show.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<!doctype html>
+<html lang="en">
+
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+  <title>API documentation</title>
+  <!-- Embed elements Elements via Web Component -->
```
Comment (Nihisil) [inline {{cookiecutter.app_name}}/lib/api/docs/views/openapi/show.html]: maybe 
```suggestion
  <!-- Embed elements via Web Component -->
```

File: {{cookiecutter.app_name}}/lib/api/docs/routers/router.go (hunk 1 above)
Comment (suho) [reply {{cookiecutter.app_name}}/lib/api/docs/routers/router.go]: Fixed in https://github.com/nimblehq/gin-templates/pull/107/commits/05db79f40c33bb4f2b77d19e3a98d95e534748bd

That's typo. And I follow the existing naming for that function. Any suggestion from you?

Comment (Nihisil) [reply {{cookiecutter.app_name}}/lib/api/docs/routers/router.go]: Just `Routes` sounds better for me, but it is subjective

**Review by Nihisil — APPROVED** [review]

File: hooks/post_gen_project.py [hunk 3]
```diff
@@ -101,6 +101,7 @@ def init_git(message):
 
     # docs folder
     remove_files("docs")
+    remove_files("lib/api/docs")
```
Comment (carryall) [inline hooks/post_gen_project.py]: Should we remove the `lib/api` entirely? I tried generating the `Web only` variant and the `api` folder is still there 💭 

```suggestion
    remove_files("lib/api")
```

Screenshot 2566-08-25 at 17 56 46
Screenshot 2566-08-25 at 17 57 04

**Review by carryall — COMMENTED** [review]
I added only one comment, the rest look good 👍🏼

File: {{cookiecutter.app_name}}/lib/api/docs/routers/router.go (hunk 1 above)
Comment (suho) [reply {{cookiecutter.app_name}}/lib/api/docs/routers/router.go]: I think it's better to create an issue for this naming, it's better to have feedback from others.

File: hooks/post_gen_project.py (hunk 3 above)
Comment (suho) [reply hooks/post_gen_project.py]: Fixed in https://github.com/nimblehq/gin-templates/pull/107/commits/78e400bc8a3087fb7b483ad7c2a860c1b896e4a6

File: {{cookiecutter.app_name}}/lib/api/docs/routers/router.go (hunk 1 above)
Comment (suho) [reply {{cookiecutter.app_name}}/lib/api/docs/routers/router.go]: I created an issue for this

Comment (Nihisil) [reply {{cookiecutter.app_name}}/lib/api/docs/routers/router.go]: thanks

**Review by carryall — APPROVED** [review]

**Pull request merged**